### ISSUE TITLE
Azure Pipelines hook not needed for manual builds; master label misapplied

### ISSUE THREAD

**Issue by foolip** [web-platform-tests/wpt.fyi#1114]
https://dev.azure.com/web-platform-tests/wpt/_build/results?buildId=5920 was a manually triggered build and results showed up in both staging and prod even though the wpt.fyi hook was not invoked:
https://staging.wpt.fyi/results/?run_id=4837906225037312
https://wpt.fyi/results/?run_id=6047652466655232

This seems to be because a check run was created by Azure Pipelines, even though the run wasn't triggered by it:
https://github.com/web-platform-tests/wpt/runs/58836581

It seems like this also happened for a manual run triggered on the epochs/daily branch:
https://github.com/web-platform-tests/wpt/runs/58386076

However, for a scheduled Safari run, the check seems to not have been updated:
https://github.com/web-platform-tests/wpt/runs/58875589

@thejohnjansen @mustjab do you know if there's a difference in Azure Pipelines reporting back to GitHub depending on whether a build was triggered by push, scheduled, or manual? It seems like scheduled runs are the outlier here, not updating check runs.

@lukebjerring the consequence for wpt.fyi is that any runs triggered by push or manually on Azure Pipelines may have been picked up and labeled as master. https://wpt.fyi/results/?run_id=4797561198084096 is one such mislabeled run, but before closing this issue we'll have to remove the master label from any others that are also wrong.

**Comment by foolip** [comment]
I think https://wpt.fyi/results/?run_id=4797561198084096 might be the only mislabel run on master so far, if this Python script can be trusted to list all commits of runs labeled master from azure:
```python
import requests
r=requests.get('https://wpt.fyi/api/runs?label=azure&label=master&max-count=1000')
for run in r.json():
    print(run['full_revision_hash'])
```

**Comment by foolip** [comment]
All runs with product edge_webdriver would also need to be vetted or just deleted, most of them haven't come from master.

**Comment by lukebjerring** [comment]
Fixed by/related to #1167 #1195 #1192 #1180
